### PR TITLE
IC-416 Add handler and query to retrieve data

### PR DESCRIPTION
**Pull request by LoFrance**
With this PR we can retrieve data from database and serve with handler.

### CODE REVIEW

File: GetLatestMigrationsStatus/index.ts [hunk 1]
```diff
@@ -2,27 +2,18 @@ import express = require("express");
...
-  {
-    clientId: config.APIM_CLIENT_ID,
-    secret: config.APIM_SECRET,
-    tenantId: config.APIM_TENANT_ID
-  },
-  config.APIM_SUBSCRIPTION_ID
-);
+const client = getPool(config);
 
 const setupExpress = (): express.Express => {
```
Comment (fabriziopapi) [inline GetLatestMigrationsStatus/index.ts:12]: Why we do not apply standard express security configuration via @pagopa/io-functions-commons/dist/src/utils/express => secureExpressApp ?

File: GetLatestMigrationsStatus/handler.ts [hunk 2]
```diff
@@ -65,21 +86,51 @@ export const createSqlStatus = (dbConfig: IDecodableConfigPostgreSQL) => (
     .as("t")
     .toQuery() as NonEmptyString;
 
-export const createHandler = (): Handler => (): ReturnType<Handler> =>
+export const getLatestMigrationByOrganizationFiscalCode = (
```
Comment (balanza) [inline GetLatestMigrationsStatus/handler.ts:89]: Please add a comment line about what these small functions do, it helps to read the overall procedure

File: GetLatestMigrationsStatus/handler.ts [hunk 3]
```diff
@@ -65,21 +86,51 @@ export const createSqlStatus = (dbConfig: IDecodableConfigPostgreSQL) => (
...
-    TE.throwError<
-      string,
-      IResponseSuccessJson<{ readonly data: MigrationsStatus }>
-    >("To be Implementend"),
-    TE.mapLeft(ResponseErrorInternal),
+    getLatestMigrationByOrganizationFiscalCode(
+      config,
+      pool
+    )(organizationFiscalCode),
+    TE.mapLeft(flow(E.toError, e => ResponseErrorInternal(e.message))),
```
Comment (balanza) [inline GetLatestMigrationsStatus/handler.ts:124]: This `mapLeft` takes a `IDbError` object. Do you really need to convert it into an `Error`  first and then into a `ResponseErrorInternal`? Being the incoming value strongly typed, you can skip the `Error` and just compose the error message by yourself.

More: are you sure you aren't losing any information by using the generic `toError` helper? Does the generated `Error` contains in its message all info from `IDbError` that may be useful for troubleshooting? What does the `Error` contains if you pass a `IDbError` to `toError`? (my bet: `[Object object]`).

File: GetLatestMigrationsStatus/handler.ts [hunk 4]
```diff
@@ -65,21 +86,51 @@ export const createSqlStatus = (dbConfig: IDecodableConfigPostgreSQL) => (
...
+
+export const processResponseFromLatestMigrationResultSet = (
+  resultSet: LatestMigrationResultSet
+): TE.TaskEither<
+  IResponseErrorInternal | IResponseErrorNotFound,
+  IResponseSuccessJson<MigrationsStatus>
+> =>
+  pipe(
+    resultSet,
+    LatestMigrationResultSet.decode,
```
Comment (balanza) [inline GetLatestMigrationsStatus/handler.ts:109]: Why the need to decode an object of type `LatestMigrationResultSet` with a codec of the very same type? You either:
* don't trust the function signature (why?)
* don't need such decode

**Review by balanza — CHANGES_REQUESTED** [review]

File: GetLatestMigrationsStatus/index.ts (hunk 1 above)
Comment (LoFrance) [reply GetLatestMigrationsStatus/index.ts:13]: Thanks, fix with 7da83df

File: GetLatestMigrationsStatus/handler.ts (hunk 3 above)
Comment (LoFrance) [reply GetLatestMigrationsStatus/handler.ts]: Maybe this can be an improvement 32c96c7?

Comment (balanza) [reply GetLatestMigrationsStatus/handler.ts]: You don't need to pipe the value, just use it
```ts
    TE.mapLeft(error =>
      ResponseErrorInternal(
        `getLatestMigrationByOrganizationFiscalCode|${error.kind}: ${error.message}`
      )
    ),
```

**Review by balanza — APPROVED** [review]

**Pull request merged**